core/inst_handlers/v/RvviaInsts.cpp

```diff
@@ -13,81 +11,71 @@ namespace atlas
+    Action::ItrType viavvHelper(AtlasState* state, Action::ItrType action_it)
     {
         const AtlasInstPtr & inst = state->getCurrentInst();
-        VectorState* vector_state = state->getVectorState();
-        uint8_t const vlenb = sizeof(VLEN);
-        uint8_t const sewb = sizeof(SEW);
-        uint8_t const vl = vector_state->getVL();
-        uint8_t const vstart = vector_state->getVSTART();
-        uint32_t const rs1 = inst->getRs1();
-        uint32_t const rs2 = inst->getRs2();
-        uint32_t const rd = inst->getRd();
-        if (vstart >= vl)
-        {
-            return ++action_it;
-        }
-        VLEN const rs1_val = READ_VEC_REG<VLEN>(state, rs1 + vstart / (vlenb / sewb));
-        VLEN const rs2_val = READ_VEC_REG<VLEN>(state, rs2 + vstart / (vlenb / sewb));
-        VLEN rd_val = READ_VEC_REG<VLEN>(state, rd + vstart / (vlenb / sewb));
-        uint8_t const stop = std::min<uint8_t>(vl, (vstart / (vlenb / sewb) + 1) * (vlenb / sewb));
-        uint8_t index = vstart;
-        for (; index < stop; ++index)
-        {
-            SEW result = OP()(*(reinterpret_cast<const uint8_t*>(
-                                  rs1_val.data() + (index % (vlenb / sewb)) * sewb)),
-                              *(reinterpret_cast<const uint8_t*>(
-                                  rs2_val.data() + (index % (vlenb / sewb)) * sewb)));
-            memcpy(rd_val.data() + (index % (vlenb / sewb)) * sewb, &result, sewb);
-        }
-        WRITE_VEC_REG<VLEN>(state, rd + vstart / (vlenb / sewb), rd_val);
-        vector_state->setVSTART(index);
-        if (index != vl)
+        const MaskElements mask_elems{state, state->getVectorConfig(), atlas::V0};
+        MaskElements::MaskBitIterator mask_iter_vs1{&mask_elems,
+                                                    state->getVectorConfig()->getVSTART()};
+        MaskElements::MaskBitIterator mask_iter_vs2{&mask_elems,
+                                                    state->getVectorConfig()->getVSTART()};
+        MaskElements::MaskBitIterator mask_iter_vd{&mask_elems,
+                                                   state->getVectorConfig()->getVSTART()};
+        MaskElements::MaskBitIterator* mask_iter_vs1_ptr = inst->getVM() ? nullptr : &mask_iter_vs1;
+        MaskElements::MaskBitIterator* mask_iter_vs2_ptr = inst->getVM() ? nullptr : &mask_iter_vs1;
+        MaskElements::MaskBitIterator* mask_iter_vd_ptr = inst->getVM() ? nullptr : &mask_iter_vd;
+        Elements<Element<ElemWidth>> elems_vs1{state, state->getVectorConfig(), inst->getRs1(),
+                                               mask_iter_vs1_ptr};
+        Elements<Element<ElemWidth>> elems_vs2{state, state->getVectorConfig(), inst->getRs2(),
+                                               mask_iter_vs2_ptr};
+        Elements<Element<ElemWidth>> elems_vd{state, state->getVectorConfig(), inst->getRd(),
+                                              mask_iter_vd_ptr};
```

Comment:
Many dereferences to the same object in this block.  Create local pointers to save time (esp in debug):
```
    const auto vect_config = state->getVectorConfig();
    const auto vstart = vect_config->getVSTART();
   ...etc
```